internal/resources/azure/sql_managed_instance.go

```diff
@@ -0,0 +1,196 @@
+// More resource information here: https://azure.microsoft.com/en-gb/products/azure-sql/managed-instance/
+// Pricing information here: https://azure.microsoft.com/en-gb/pricing/details/azure-sql-managed-instance/single/
+type SQLManagedInstance struct {
+	Address            string
+	Region             string
+	SKU                string
+	LicenceType        string
+	Cores              *int64
+	StorageSizeInGb    *int64
+	StorageAccountType string
+	// LongTermRetentionStorageGB defines a usage param that allows users to define how many gb of cold storage the database uses.
+	// This is storage that can be kept for up to 10 years.
+	LongTermRetentionStorageGB *int64 `infracost_usage:"long_term_retention_storage_gb"`
+	BackupStorageGb            *int64 `infracost_usage:"backup_storage_gb"`
+}
+
+// PopulateUsage parses the u schema.UsageData into the SQLManagedInstance.
+// It uses the `infracost_usage` struct tags to populate data into the SQLManagedInstance.
+func (r *SQLManagedInstance) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+// BuildResource builds a schema.Resource from a valid SQLManagedInstance struct.
+// This method is called after the resource is initialised by an IaC provider.
+// See providers folder for more information.
+func (r *SQLManagedInstance) BuildResource() *schema.Resource {
+	costComponents := r.costComponents()
+
+	return &schema.Resource{
+		Name: r.Address,
+		UsageSchema: []*schema.UsageItem{
+			{Key: "backup_storage_gb", DefaultValue: 0, ValueType: schema.Int64},
+			{Key: "long_term_retention_storage_gb", DefaultValue: 0, ValueType: schema.Int64},
+		},
+		CostComponents: costComponents,
+	}
+}
+
+func (r *SQLManagedInstance) costComponents() []*schema.CostComponent {
+
+	costComponents := []*schema.CostComponent{
+		{
+			Name:           fmt.Sprintf("Compute (%s %s Cores)", strings.ToTitle(r.SKU), strconv.FormatInt(*r.Cores, 10)),
+			Unit:           "hours",
+			UnitMultiplier: decimal.NewFromInt(1),
+			HourlyQuantity: decimalPtr(decimal.NewFromInt(1)),
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr(vendorName),
+				Region:        strPtr(r.Region),
+				Service:       strPtr(sqlMIServiceName),
+				ProductFamily: strPtr(sqlMIProductFamily),
+				AttributeFilters: ([]*schema.AttributeFilter{
+					{Key: "productName", Value: r.productDescription()},
+					{Key: "skuName", Value: r.meteredName()},
+				}),
+			},
+			PriceFilter: priceFilterConsumption,
+		},
+	}
+
+	if r.BackupStorageGb != nil {
+		costComponents = append(costComponents, r.sqlMIStorageCostComponent(), r.sqlMIBackupCostComponent())
+	}
+
+	if r.LicenceType == "LicenseIncluded" {
+		costComponents = append(costComponents, r.sqlMILicenseCostComponent())
+	}
+	if r.LongTermRetentionStorageGB != nil {
+		costComponents = append(costComponents, r.sqlMILongTermRetentionStorageGBCostComponent())
+	}
+	return costComponents
+}
+
+func (r *SQLManagedInstance) productDescription() *string {
+
```

Comment:
there's a couple of occasion where there is an newline after the function name - could you remove this please